onnxscript/irbuilder.py

```diff
@@ -120,14 +125,48 @@ def __str__(self):
         return helper.printable_attribute(self.attr_proto)
 
     @property
-    def name(self):
+    def name(self) -> str:
         return self.attr_proto.name
 
     @property
-    def type(self):
+    def type(self) -> onnx.AttributeProto.AttributeType:
         return self.attr_proto.type
 
 
+@dataclasses.dataclass(frozen=True)
+class IRAttributeParameter:
+    """An attribute parameter (representing a formal parameter).
+
+    It may or may not carry a default value.
+
+    Attributes:
```

Comment:
Is this out-dated, or is this the accepted convention? That is, should this list be "name, type, default_value" or as below?

Reply:
I should add the other attributes. thanks for catching

Reply:
Done

Reply:
![image](https://user-images.githubusercontent.com/11205048/232101208-57af4913-f3ed-4a5a-a845-b55b0145372e.png)

Is this the same view you saw?

Reply:
Yes

Reply:
Wait so was this good or did you mean something else?

Reply:
I guess I am trying to understand the conventions here. The current form makes sense as the "public interface" exposed. Technically, however, `default_value` is also an attribute. Is the idea that it is "private"? Would renaming `default_value` to `_default_value` be appropriate?

Reply:
Added default_value. Thanks!